tests/acceptance/01_vars/04_containers/multiple_cmerge_variable_dynamic.cf

```diff
@@ -0,0 +1,162 @@
+# This policy is synthetic, but represents a real life pattern where data is
+# published from multiple bundles and then merged before use.
+# Note: The multiple_cmerge_variable_static.cf test is nearly identical. It simply uses hard coded variables instead of variable variables.
+
+#@@ -97,7 +102,7 @@
+# 
+#   vars:
+#       "cron"
+#-        data => readjson( "$(this.promise_filename).policy_1.cron.json"),
+#+        data => readjson( "$(this.promise_filename).$(this.bundle).$(this.promiser).json"),
+#         meta => { "cron_rules" };
+# 
+#        "s" string => format( "%S", cron );
+#@@ -120,7 +125,7 @@
+#       # # For example: Cron entries
+#        "cron"
+# 
+#-         data => readjson( "$(this.promise_filename).policy_2.cron.json"),
+#+         data => readjson( "$(this.promise_filename).$(this.bundle).$(this.promsier).json"),
```

Comment:
this.promiser refers to the object making the promise. In this case its cron.

https://docs.cfengine.com/lts/reference-special-variables-this.html#this-promiser